### ISSUE TITLE
Docker: Internal server error

### ISSUE THREAD

**Issue by xyzulu** [imapsync/imapsync#429]
Consistently I see this error (different servers as well)

> ++++ Looping on each one of 9 folders to sync
> ETA: Tuesday 12 December 2023-12-12 02:45:59 +0000 UTC  0 s  3430/3430 msgs left
> Folder     1/9 [Archive]                           -> [Archive]                          
> Host1: folder [Archive] has 6 messages in total (mentioned by SELECT)
> Host2: folder [Archive] has 6 messages in total (mentioned by SELECT)
> Host2: folder [Archive] permanentflags: 
> Host1: folder [Archive] considering 6 messages
> Host2: folder [ArchHTTP/1.0 500 Internal Server Error
> Date: Tue Dec 12 02:47:00 2023 GMT
> Connection: close
> Server: Net::Server::HTTP/2.013
> Content-type: text/html
> 
> <html>
> <body>
> <h1>Internal Server Error</h1><p>Server Timeout on process
> </p></body>
> </html>

I don't have anything special.. running a stock docker setup. It seems the 500 error is being generated here:
POST /cgi-bin/imapsync HTTP/1.1" 500

Any ideas?

This doesn't happen if I use the online interface at: https://imapsync.lamiral.info/X/

**Comment by xyzulu** [comment]
That makes sense.. thanks!

**Comment by gilleslamiral** [comment]
Update made, have a go

     docker pull gilleslamiral/imapsync

**Comment by xyzulu** [comment]
That has fixed the 500 error for sure.. I saw a process run for over 4 mins this time. Still, the docker implementation of https://imapsync.lamiral.info/X/ in my case doesn't give the usual 6s feedback. Yet, if I run the same sync with the same container on the same hardware via the cli.. I get plenty of feedback (visual progress).

It's not my specialty, but perhaps another web server ie nginx might be better in the container?? I think the web server part here is not optimal.

**Comment by gilleslamiral** [comment]
With the Net::Server::HTTP server, the running feedback interval is not based on time, it is every 4 Kb of output. It's not my decision, I discovered it as you did. Anyway, the imap job is done, so that's not a first bug issue for me, even I did spend hours on it, just to understand where it comes from. I'll post a feature request to Net::Server::HTTP maintainers.

For Ngnix, I just know its name so not for now.
It could be Apache, I know it quite well, but I still have to learn, play and master the Docker way of it.

Caddy? I just learned the name...
